src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -248,59 +238,46 @@ private void importRoslynReport(String reportPath, Issuable projectIssuable) {
       }
     }
 
-    JsonParser parser = new JsonParser();
-    JsonElement issues = parser.parse(contents).getAsJsonObject().get("issues");
-    if (issues != null) {
-      for (JsonElement issueElement : issues.getAsJsonArray()) {
-        JsonObject issue = issueElement.getAsJsonObject();
+    new SarifParser(new SarifParserCallback() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)